### PR TITLE
oak_runtime: Return a `Result` from load_wasm

### PR DESCRIPTION
**Pull request by blaxill**
#564

### Checklist

- [X] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [X] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak/server/rust/oak_runtime/src/runtime.rs [hunk 1]
```diff
@@ -67,7 +67,10 @@ impl Runtime {
                     }
                     Some(NodeConfiguration_oneof_config_type::wasm_config(
                         proto::manager::WebAssemblyConfiguration { module_bytes, .. },
-                    )) => load_wasm(&module_bytes),
+                    )) => load_wasm(&module_bytes).map_err(|e| {
+                        error!("Error loading wasm: {}", e);
```
Comment (tiziano88) [inline oak/server/rust/oak_runtime/src/runtime.rs]:
```suggestion
                        error!("Error loading Wasm module: {}", e);
```

File: oak/server/rust/oak_runtime/src/node/mod.rs [hunk 2]
```diff
@@ -41,15 +41,31 @@ pub enum NodeConfiguration {
...
+            ConfigurationError::WasmiModuleInializationError(e) => {
+                write!(f, "Failed to initialize wasmi::Module: {}", e)
+            }
+        }
     }
 }
 
+/// Loads a Wasm module into a node configuration, returning an error if `wasmi` failed to load the
+/// module.
+pub fn load_wasm(wasm_bytes: &[u8]) -> Result<NodeConfiguration, ConfigurationError> {
```
Comment (tiziano88) [inline oak/server/rust/oak_runtime/src/node/mod.rs]: Why not return `Result<NodeConfiguration, wasmi::Error>` instead, which is more precise?

Comment (blaxill) [reply oak/server/rust/oak_runtime/src/node/mod.rs]: I think we will want to catch more configuration errors in the future, when there are more nodes. Perhaps `Runtime::configure_and_run` returning `Result<(RuntimeRef, ChannelWriter), ConfigurationError>`, although maybe adding the enum now is a bit premature.

File: oak/server/rust/oak_runtime/src/runtime.rs (hunk 1 above)
Comment (blaxill) [reply oak/server/rust/oak_runtime/src/runtime.rs]: Done

**Pull request merged**